### DISCUSSION TITLE
Turning off gateway options in DHCP when operating Pico W in AP mode.

### DISCUSSION THREAD

**Discussion by the-maldridge** [earlephilhower/arduino-pico#2014] [Q&A]
The title pretty much sums it up, but I'm trying to figure out how to suppress handing out a default gateway when operating in AP mode.  I can see in https://github.com/earlephilhower/arduino-pico/blob/master/libraries/WiFi/src/dhcpserver/dhcpserver.c where the options get defined and written, but I don't see any entrypoints to that code to influence what options get sent.  I'm ultimately trying to work out how to disable both the gateway option and the DNS option, since I want to be able to connect to a network the Pico is presenting from a device with two PHYs and if the Pico offers these parameters it will interfere with the working network configuration on such a host.

**Comment by earlephilhower** [comment]
There are no APIs presently to support this, but if you want to try your hand at defining them and making a PR I'd be happy to take a look at it!  Probably just need to isolate https://github.com/earlephilhower/arduino-pico/blob/c670f66140e4188aab4a6f1a92644791a89cb153/libraries/WiFi/src/dhcpserver/dhcpserver.c#L278-L279

**Reply by the-maldridge** [reply]
Hmm, I'm happy to try my hand at making a PR.  Do you have strong opinions on how you might want this extension of the API to look?

**Reply by earlephilhower** [reply]
There's already [` WiFi.config()` call](https://github.com/earlephilhower/arduino-pico/blob/c670f66140e4188aab4a6f1a92644791a89cb153/libraries/WiFi/src/WiFiClass.h#L193) which lets you set GW and DNS, maybe that could be used/overridden here to keep things symmetric?

**Reply by the-maldridge** [reply]
I've got a patch in progress, but I'm running into trouble testing it.  I followed the instructions to reinstall everything from git, but the arduino-cli is now complaining that it can't find a platform `rp2040:rp2040` installed.  Is there some extra step after cloning everything into the right place to get arduino-cli to see the fork of the platform?
